restapi/src/main/java/io/stargate/web/docsapi/service/util/DocsApiUtils.java

```diff
@@ -233,6 +233,17 @@ public static String convertJsonToBracketedPath(String path) {
     return newPath.toString();
   }
 
+  public static String[] bracketedPathAsArray(String bracketedPath) {
+    String[] split = bracketedPath.split("\\]\\[");
+    if (split.length > 0) {
+      split[0] = split[0].substring(2);
```

Comment:
should we immediately do a split on the `bracketedPath.substring(2, bracketedPath.length -1)` and thus avoid this gymnastics here?

Reply:
The idea here is that the input string could be something like `$['a'][0]['b']` and I want an array with values ['a', 0, 'b']` from it.

So the initial split gets me a weird looking array with `["$['a'", "0", "'b']"]`. Then I have to clean up the first and last elements of the array by chopping off the first two characters of the first element `$['a'` and the final character of the last element `b'`. So I think the two steps here might be necessary